packages/lib/src/vis/shared/AdaptativeThresholdEnforcer.tsx

```diff
@@ -0,0 +1,18 @@
+import { useRaycasterThreshold } from '../hooks';
+
+interface Props {
+  lineThreshold?: number;
+  pointThreshold?: number;
+}
+
+function AdaptativeThresholdEnforcer(props: Props) {
+  const { lineThreshold = 1, pointThreshold = 1 } = props;
+
+  useRaycasterThreshold('Line', lineThreshold);
+  useRaycasterThreshold('Points', pointThreshold);
```

Comment:
Would it make sense to move the content of the `useRaycastThreshold` hook in here, change `Props` to `RaycasterParams` and just re-assign the entire `raycaster.params` object?

Reply:
And in the `useFrame`, I guess, assign the new threshold to any key that exists in the `RaycasterParams` object.

Reply:
Yeah I thought about it but this forces me to dig into the `raycaster.params` (and deal with several levels of `undefined`) to find the `desiredThreshold` values...

Reply:
Ah right, didn't realise `RaycasterParameters` had potentially more stuff than just thresholds. In that case I would maybe define an interface `Thresholds { Line?: number; Points?: number }` and work with that, still inline inside the component instead of in a custom hook. It's fine if you have to duplicate lines IMO.

Reply:
The interface is generic but you can only specify `threshold` through it from what I get: https://threejs.org/docs/#api/en/core/Raycaster.params